app/src/main/java/com/andrew/tasky/presentation/agenda_item_detail/AgendaItemDetailFragment.kt

```diff
@@ -240,10 +262,77 @@ class AgendaItemDetailFragment : Fragment(R.layout.fragment_agenda_item_detail)
             descriptionTextView.isClickable = isEditing
 
             time.isClickable = isEditing
+            timeButton.isVisible = isEditing
+            timeButton.isClickable=isEditing
             date.isClickable = isEditing
+            dateButton.isVisible = isEditing
+            dateButton.isClickable = isEditing
 
+            reminderTextView.isClickable = isEditing
             reminderButton.isClickable = isEditing
             reminderButton.isVisible = isEditing
         }
     }
+
+    //View parameter determines which view popup appears under
+    private fun showReminderOptionsPopupMenu(view: View){
+        val popupMenu = PopupMenu(requireContext(), view)
+        popupMenu.inflate(R.menu.menu_reminder_time_options)
+        binding.reminderTextView.apply {
+            popupMenu.setOnMenuItemClickListener {
+                when (it.itemId) {
+                    R.id.tenMinutes -> {
+                        text = getString(R.string.ten_minutes_before)
+                        true
+                    }
+                    R.id.thirtyMinutes -> {
+                        text = getString(R.string.thirty_minutes_before)
+                        true
+                    }
+                    R.id.oneHour -> {
+                        text = getString(R.string.one_hour_before)
+                        true
+                    }
+                    R.id.sixHours -> {
+                        text = getString(R.string.six_hours_before)
+                        true
+                    }
+                    R.id.oneDay -> {
+                        text = getString(R.string.one_day_before)
+                        true
+                    }
+                    else -> true
+                }
+            }
+        }
+        popupMenu.show()
+    }
+
+    private fun showDeleteDialog(){
+        val deleteDialog = Dialog(requireActivity())
```

Comment:
I'd recommend to make this a `DialogFragment` instead (and put its logic into its own subclass, so you can easily reuse it). That's effectively a fragment with the same lifecycle a fragment has, you can use ViewModels there and it's also not dismissed on screen rotations like the normal Dialog